[16.0][IMP] hr_employee_medical_examination: Add Activities filter + kanban view

victoralmau opened the pull request:
Add Activities filter + kanban view

**Before**
![antes](https://github.com/user-attachments/assets/fd677c30-7d9f-4fa4-bda8-f191d747a828)

**After**
![despues](https://github.com/user-attachments/assets/a673e17c-2024-475f-8dee-441a9529f52d)

Please @pedrobaeza and @carlos-lopez-tecnativa can you review it?

@Tecnativa TT57402

victoralmau committed:
[IMP] hr_employee_medical_examination: Add Activities filter + kanban view

TT57402

carlos-lopez-tecnativa commented on hr_employee_medical_examination/views/hr_employee_medical_examination_views.xml:
```diff
@@ -113,13 +132,58 @@
+                                            /></span>
+                                        <span class="o_kanban_record_subtitle"><field
+                                                name="state"
+                                            /></span>
```
Could you use `widget="label_selection"` and add some classes to differentiate the states based on colors?
<img width="1378" height="343" alt="image" src="https://github.com/user-attachments/assets/dcafe3b0-7d3a-4f80-86d0-0950b2a93410" />

Like as 
<img width="1378" height="343" alt="image" src="https://github.com/user-attachments/assets/8b9388d5-13be-455d-9161-581b75218269" />

victoralmau replied:
Thanks.

<img width="836" height="145" alt="ejemplo" src="https://github.com/user-attachments/assets/bbd19b97-108e-4e9f-91bf-04b456886236" />

carlos-lopez-tecnativa reviewed (changes requested):
Could you improve the model description to remove “HR” and keep only “Employee Medical Examination”? I understand that activities read the [model description](https://github.com/odoo/odoo/blob/1adee2b7452c072bcce51cd75641a9e4c2b7f219/addons/mail/models/res_users.py#L213-L220).

https://github.com/Tecnativa/hr/blob/768c50a810ebbcfc3a66743d696cbdd565531473/hr_employee_medical_examination/models/hr_employee_medical_examination.py#L12
<img width="381" height="168" alt="image" src="https://github.com/user-attachments/assets/850bc27c-ff31-4138-acee-f1c6b8d53332" />

victoralmau committed:
[IMP] hr_employee_medical_examination: Edit the model description to remove "HR"

Related to https://github.com/OCA/hr/pull/1490#pullrequestreview-3068562926

TT57402

carlos-lopez-tecnativa reviewed (approved):
LGTM

pedrobaeza commented on hr_employee_medical_examination/views/hr_employee_medical_examination_views.xml:
```diff
@@ -113,13 +132,62 @@
         <field name="name">Employee Medical Examinations</field>
         <field name="res_model">hr.employee.medical.examination</field>
-        <field name="view_mode">tree,form</field>
+        <field name="view_mode">kanban,tree,form</field>
```
Why `kanban` is before `tree` one?

victoralmau replied:
Initially, I thought that the kanban view was more useful, which is why I defined it first, but your question made me think that it might cause more confusion than improvement, so I changed it so that it is not defined first.

pedrobaeza reviewed (approved):
/ocabot merge minor
